src/instana/instrumentation/django/middleware.py

```diff
@@ -1,242 +1,260 @@
+                        agent.options.secrets_matcher,
+                        agent.options.secrets_list,
+                    )
+                    request.span.set_attribute("http.params", scrubbed_params)
+                if "HTTP_HOST" in env:
+                    request.span.set_attribute("http.host", env["HTTP_HOST"])
+            except Exception:
+                logger.debug("Django middleware @ process_request", exc_info=True)
+
+        def process_response(
+            self, request: Type["HttpRequest"], response: "HttpResponse"
+        ) -> "HttpResponse":
+            try:
+                if request.span:
+                    if 500 <= response.status_code:
+                        request.span.assure_errored()
+                    # for django >= 2.2
+                    if request.resolver_match is not None and hasattr(
+                        request.resolver_match, "route"
+                    ):
+                        path_tpl = request.resolver_match.route
+                    # django < 2.2 or in case of 404
+                    else:
+                        try:
+                            from django.urls import resolve
+
+                            view_name = resolve(request.path)._func_path
+                            path_tpl = "".join(url_pattern_route(view_name))
+                        except Exception:
+                            # the resolve method can fire a Resolver404 exception, in this case there is no matching route
+                            # so the path_tpl is set to None in order not to be added as a tag
+                            path_tpl = None
+                    if path_tpl:
+                        request.span.set_attribute("http.path_tpl", path_tpl)
+
+                    request.span.set_attribute(
+                        SpanAttributes.HTTP_STATUS_CODE, response.status_code
+                    )
+                    if hasattr(response, "headers"):
+                        self._extract_custom_headers(
+                            request.span, response.headers, format=False
+                        )
+                    tracer.inject(request.span.context, Format.HTTP_HEADERS, response)
+            except Exception:
+                logger.debug("Instana middleware @ process_response", exc_info=True)
+            finally:
+                if hasattr(request, "span") and request.span:
+                    if request.span.is_recording():
+                        request.span.end()
+                    request.span = None
+                if hasattr(request, "token") and request.token:
+                    context.detach(request.token)
+                    request.token = None
+            return response
+
+        def process_exception(
+            self, request: Type["HttpRequest"], exception: Exception
+        ) -> None:
+            from django.http.response import Http404
+
+            if isinstance(exception, Http404):
+                return None
```

Comment:
This change was made 3 years ago in [this](https://github.com/instana/python-sensor/pull/294) PR because our spec advises to not record 4xx status codes as exceptions [here](https://github.ibm.com/instana/technical-documentation/blob/master/tracing/specification/README.md#capturing-errors-on-http-spans).